go.mod

```diff
@@ -2,51 +2,52 @@ module github.com/pulumi/pulumi-yaml
 
 go 1.19
 
+replace github.com/hashicorp/vault/api v1.8.2 => github.com/hashicorp/vault/api v1.9.0
+
+replace google.golang.org/protobuf v1.33.0 => google.golang.org/protobuf v1.31.0
+
 require (
 	github.com/blang/semver v3.5.1+incompatible
 	github.com/ettle/strcase v0.1.1
-	github.com/golang/protobuf v1.5.4
+	github.com/golang/protobuf v1.5.3
 	github.com/google/shlex v0.0.0-20191202100458-e7afc7fbc510
 	github.com/hashicorp/go-multierror v1.1.1
 	github.com/hashicorp/hcl/v2 v2.17.0
-	github.com/hexops/autogold v1.3.0
 	github.com/iancoleman/strcase v0.2.0
 	github.com/pkg/errors v0.9.1
-	github.com/pulumi/pulumi/pkg/v3 v3.113.0
-	github.com/pulumi/pulumi/sdk/v3 v3.113.0
+	github.com/pulumi/pulumi/pkg/v3 v3.112.1-0.20240415121624-fc23f9bb0851
+	github.com/pulumi/pulumi/sdk/v3 v3.112.1-0.20240415121624-fc23f9bb0851
```

Comment:
Just waiting until v3.114.0 gets released, which includes https://github.com/pulumi/pulumi/pull/15936, which is necessary for the tests to work.